### PR TITLE
feat(PolicyDetails): RHINENG-10399 - Migrate Policy Rules Tab to API v2

### PR DESCRIPTION
**Pull request by bastilian**
This migrates the Policy Details Rules Tab to use the REST API v2.
It also introduces the treeTable functionality, it can and should be enabled for testing, but is disabled by default, until RHINENG-13519 is resolved:

**How to test:**

1) Enable REST API v2 & AsyncTable
2) Open the Policy details rules tab
3) Verify the rules are shown for each minor OS profile and rules filter, paginate and sort correctly.
4) Enable the ruleTree by passing on the `ruleTree` returned by `data` in the `TailoringsTab` component and pass it on to the rules table.
5) Verify the tree expands correctly, and loads the rules and rule details (These might not be the correctly assigned rules yet due to API changes still needed)

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [X] General Coding Practices

### CODE REVIEW

**Review by gkarat — COMMENTED** [review]
A quick look:

1. All three filters add chips but don’t filter the table,
2. I tried to open PCI-DSS v4.0 Control Baseline for Red Hat Enterprise Linux 7 policy and checked the Verify User Who Owns shadow File rule: Ansible remediation is available, but it’s marked as Manual in the Remediation column

negligible:

1. Sorting by name is set by default: is is intended? The previous implementation didn’t sort the table
2. While sorting, no loading indicator

**Review by LightOfHeaven1994 — COMMENTED** [review]
I took a note about not working things that you mentioned, I also found some things:

1. We miss export button
2. We miss Edit rules button
3. No need to request rule on expanding row
4. We miss value editing feature. I took a look how it should work and seems we will need to do N backend request when expanding row that has value_checks array non empty. So I believe we will need to check what rules is expanded and see from `policies/id/tailorings/id/rules` if our rule has `value_checks` non empty - if so it means that we have value editing available. We will need to do backend request on `security_guides/id/value_definitions/id` to grab Name of value definition. And to show actual value - we will grab it from the first request that we already made for this page - `/policies/id/tailorings/id` where we have array of `value_overrides`

**Comment by bastilian** [comment]
@LightOfHeaven1994 The issues you found should now be addressed. Everything may still a bit wonky, but we'll iron that out in following PRs.

**Review by LightOfHeaven1994 — COMMENTED** [review]
Couple things:

1. reset value doesn't work (still uses GQL), I think it can be addressed in different PR as follow up
image

2. It would be great if we don't reload the whole page when changing 1 value ( when we use GQL, page refresh is not happening)

File: src/PresentationalComponents/Tailorings/Tailorings.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  level = 0,
+  ouiaId,
+  onValueOverrideSave,
+  ...rulesTableProps
+}) => {
+  const { data, loading, error } = useTailorings({
+    params: [
+      policy.id,
+      undefined,
+      undefined,
```
[the remark this answers is not in the document]
Comment (bastilian) [reply src/PresentationalComponents/Tailorings/Tailorings.js:29]: added.

File: src/PresentationalComponents/Tailorings/hooks/useTailoringsData.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    openRuleGroups?.length > 0
+      ? `rule_group_id ^ (${openRuleGroups.map((id) => `"${id}"`).join(',')})`
+      : undefined;
+
+  const ruleParams = useMemo(
+    () => [
+      undefined,
+      // TODO this is a hack: The state value defaults should come from the state itself
+      pagination?.limit || 10,
+      pagination?.offset || 0,
```
[the remark this answers is not in the document]
Comment (bastilian) [reply src/PresentationalComponents/Tailorings/hooks/useTailoringsData.js:45]: also added.

**Comment by bastilian** [comment]
@LightOfHeaven1994 Thank you for testing again! I'll fix the remaining issues in the following task when migrating the Systems Details and Create Policy wizard.

**Review by LightOfHeaven1994 — APPROVED** [review]

**Pull request merged**